### ISSUE TITLE
Display the value of each process (manufacturing, resource, and food production) and allow player to set priority

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#611]
**Is your feature request related to a problem? **
1. Say a player has in mind to construct more buildings and expand a settlement.
2. It's quite difficult for him to fathom why a settlement would set high priority to automatically make work gloves and make compost.
3. Meanwhile, he sees there's no indication as to whether his settlement would support activities related to preparing construction materials such as cement and concrete, steel posts, etc. for 

**Describe the solution you'd like**
1. The manufacture process tab should allow users to view the priority and the modifiers that create the priority score for each process. 
2. The player should be able to adjust each process's priority manually, such as clicking on the + or - buttons.
3. For processes that are currently not possible to achieve, they should be grouped together under an unchecked list or a grey out list. 
4. If the player click on the check box, it will prompt the settlement to do all it can to be able to support this process. e.g. it can give out instructional steps, such as the engineers must be advanced to level 4, or it may say that some tech upgrade must be performed. 
5. Preferably, a list of reasons will be provided to the player to explain why it is not possible to carry out this process yet.

**Describe alternatives you've considered**
1. Use of a building window in Swing would promote more interactivity with the simulation. However, it needs more time to spend on debugging and to design more visually appealing UI.
2. Use of the console UI will be quicker but hard to code in choices for players since printing out textual information is sequential and presentation is more linear and flat. 

**Specifications  (please complete the following information):**
 - pre-3.4.0 build 6868

**Additional context**

![image](https://user-images.githubusercontent.com/1168584/177019730-ab016b72-9d58-4427-9da4-45b5aa64e31a.png)

See below. Work Gloves has 35.1 value point. Why is it being picked among other processes ? Why is there only one process to be run ?
![image](https://user-images.githubusercontent.com/1168584/177019828-138b7d0a-cccb-4934-bbc5-353fa0cefc36.png)

**Comment by mokun** [comment]
@bevans2000 

I made very small changes to these two methods, namely, getResourceManufacturingProcessDemand and
   getPartManufacturingProcessDemand) in GoodsManager in the last commit, just so you know.

**Comment by bevans2000** [comment]
Thanks I saw that. The processDemand changes I will relocate to the appropriate Good subclass.

**Comment by mokun** [comment]
@bevans2000 

This thread is actually very similar in scope with https://github.com/mars-sim/mars-sim/issues/519.

I see mars-sim is already complex enough for players to use different approach each time. 

Especially the fact that manufacturing process is where player is given the opportunity to be creative.

What can we do next to reduce the complexity of choosing what to manufacture and improve interactivity ?

**Comment by mokun** [comment]
@bevans2000 

We could provide more visibility to how a manufacturing process is chosen by showing the score and the modifiers on it. 

Can it be done in console first ?

**Comment by mokun** [comment]
@bevans2000 
So, are we by any chance converting these processes to use Rating to show players the breakdown scores from each category on how it contributes to the final process score ?

**Comment by bevans2000** [comment]
> @bevans2000 So, are we by any chance converting these processes to use Rating to show players the breakdown scores from each category on how it contributes to the final process score ?

Yes we could do. The point is to give the users (& us) better understanding why certain things are being selected. The autonomous simulation makes some odd decisions at the moment but we have no idea why. I hope showing composition of a score will help understand.

**Comment by mokun** [comment]
> showing composition of a score will help understand.

Agree. But how do you envision the Rating class be also applied to a process at this point ? 

Which one process (Manufacture, Food Production, Meal/Dessert Making, Resource/Waste, etc.) will you be most interested in implementing it as the first pilot case so that it would later be applied to all other types of processes, including others not listed above ?

**Comment by bevans2000** [comment]
Genuinely I don't know but the concept seem correct to be applied to a Process.

**Comment by mokun** [comment]
We now have a score for each resource/waste process. 

Need to standardize further for comparison.

Will work on that in 3.9.0 or 4.0.0.

**Comment by mokun** [comment]
Below is a prototype of representing the average duty cycle of a resource process in Resource Tab :  

<img width="642" height="969" alt="Image" src="https://github.com/user-attachments/assets/593e7b93-7650-47d2-86ae-d95040bf23b4" />

It gives us some ideas about how often this process is run.
 
The highest percentage is 100%. The lowest is 0%.

**Comment by bevans2000** [comment]
Ok, so how does the percentage map to how often it is run? 100% means once a day?

**Comment by mokun** [comment]
> Ok, so how does the percentage map to how often it is run? 

Currently, each time a resource process is run, it would capture that pulse's millisols.

The duty % = accumulated millisols / the total millisols since the start of the simulation

> 100% means once a day?

I00 % means it is always running and never stops once.

Let's say now is at sol 5:300. 

10% means the process is run at the average of 10% of the whole time so far.

**Comment by mokun** [comment]
@bevans2000 

I just added the orange color for the new ProcessState.LOCK_ON in ResourceProcessPanel.

As soon as the input resources are available, it will keep on running that process non-stop.

In the past, if I click on a process to change the state from yellow (idle) to green (running), because there's a timer countdown to every process, it will change it by itself.

Now it's locked in.

I'll commit this tomorrow.

<img width="553" height="809" alt="Image" src="https://github.com/user-attachments/assets/c930f6de-c700-4d6c-b925-6ab397902e43" />
